Add support for getting next extended grapheme cluster break (#361)
stdlib-js/stdlib: lib/node_modules/@stdlib/string/next-grapheme-cluster-break/README.md

<!--

@license Apache-2.0

Copyright (c) 2020 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# nextGraphemeClusterBreak

> Return the next extended grapheme cluster break in a string after a specified position.

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- Package usage documentation. -->

<section class="usage">

## Usage

```javascript
var nextGraphemeClusterBreak = require( '@stdlib/string/next-grapheme-cluster-break' );
```

> #### nextGraphemeClusterBreak( string, position )

Reviewer: Any particular reason to make `position` a required parameter, rather than optional parameter as in the original [RFC](https://github.com/stdlib-js/stdlib/issues/357)? I would think that a reasonable default `fromIndex` would be `0`.

Author: I did this because it seemed quite natural that `nextGraphemeClusterBreak` would need a position to justify the "next". Also, does this mean that `prevtGraphemeClusterBreak` would have to have a default `beforeIndex` of `str.length - 1`?

Reviewer: @ShraddheyaS Yeah, I sympathize with the conflict with `next` and `fromIndex`; however, I think that, for consistency with other similar functions, such as `indexOf`, we should have the second argument be optional.

Re: `prev*`. This is a tricky one, where I don't think there is a single "right" answer. There would be nothing "wrong" with iterating a string from start to end and invoking this API, in which case a default of `0` would be fine. Alternatively, if we believe that `prev*` is intended for end to start iteration, then `str.length-1` makes sense. I think it is fine to go with `str.length-1`.

Author: Yes, `prev*` is intended for an end to start iteration. Will make the defaults as `0` and `str.length - 1` for `next*` and `prev*` respectively.

Reviewer: Sounds good!

```javascript
var out = nextGraphemeClusterBreak( 'last man standing', 4 );
// returns 5
```

</section>

<!-- /.usage -->

<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

## Notes

-   If `string` is an empty string, the function returns `0` irrespective of `position`.

</section>

<!-- /.notes -->

<!-- Package usage examples. -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var nextGraphemeClusterBreak = require( '@stdlib/string/next-grapheme-cluster-break' );

console.log( nextGraphemeClusterBreak( 'last man standing', 4 ) );
// => 5

console.log( nextGraphemeClusterBreak( 'presidential election', 8 ) );
// => 9

console.log( nextGraphemeClusterBreak( 'अनुच्छेद', 1 ) );
// => 3

console.log( nextGraphemeClusterBreak( '🌷', 0 ) );
// => 2
```

</section>

<!-- /.examples -->

<!-- Section for describing a command-line interface. -->

* * *

<section class="cli">

## CLI

<!-- CLI usage documentation. -->

<section class="usage">

### Usage

```text

> Usage: next-grapheme-cluster-break [options] [<string>] --pos=<index>

Reviewer: If `pos` is allowed to be optional, then `pos` no longer needs to be a required option.

distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# nextGraphemeClusterBreak

> Return the next extended grapheme cluster break in a string after a specified position.

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- Package usage documentation. -->

<section class="usage">

## Usage

```javascript
var nextGraphemeClusterBreak = require( '@stdlib/string/next-grapheme-cluster-break' );
```

#### nextGraphemeClusterBreak( string\[, fromIndex] )

Returns the next extended grapheme cluster break in a string after a specified position.

```javascript
var out = nextGraphemeClusterBreak( 'last man standing' );
// returns 1
```

By default, the first extended grapheme cluster break in the string is returned. For the next extended grapheme cluster break after a specified position in the string, provide a `fromIndex`.

```javascript

> var str = nextGraphemeClusterBreak( 'last man standing', 4 );

Reviewer: ```suggestion
var out = nextGraphemeClusterBreak( 'last man standing', 4 );
```

// returns 5
```

</section>

<!-- /.usage -->

<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

## Notes

-   If `string` is an empty string, the function returns `-1` irrespective of `fromIndex`.

> -   If there is no extended grapheme cluster break after `fromIndex`, the function returns `-1`.

Reviewer: ```suggestion
-   If an extended grapheme cluster break does not exist after `fromIndex`, the function returns `-1`.
```

> -   Note that `fromIndex` does **not** refer to a visual character position, but to an index in the ordered sequence of [UTF-16][utf-16] code units.

Reviewer: **Aside**: we may want to consider, if we don't already have such functionality, one or more string utilities (as separate packages) for indexing, and maybe iterating over, visual characters. I believe such utilities would build on the current work.
